### ISSUE TITLE
[Bug]: Failed to search reason cannot find segment in QueryNode err %!w(<nil>) after scaling in querynode

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#16786]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220505-333b439a
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.0.dev39
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed with `<MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=23, reason=Search 29 failed, reason cannot find segment 432993395697778690 in QueryNode err %!w(<nil>))> ` during and after scaing in querynodes
1. Deploy milvus cluster with 5 querynodes
2. Create collection with 2 shards
3. Insert 3000*5 entities for 5 times
4. Load collection with 2 replica number
5. Start a daemon thread to continuously search during scaling, and some querynodes restart and search failed
6. After scaling, search still failed

**Server logs:**
[mic_logs.tar.gz](https://github.com/milvus-io/milvus/files/8629775/mic_logs.tar.gz)

### Steps To Reproduce

```markdown
release_name = xxx
        try:
            # connect and create
            connections.connect("default", host=host, port=19530)
            collection_w.init_collection(name=cf.gen_unique_str("scale_out"), schema=cf.gen_default_collection_schema())

            # insert 10 sealed segments
            for i in range(5):
                df = cf.gen_default_dataframe_data(start=i * ct.default_nb)
                collection_w.insert(df)
                assert collection_w.num_entities == (i + 1) * ct.default_nb

            collection_w.load(replica_number=2)

            @counter
            def do_search():
                """ do search """
                search_res, is_succ = collection_w.search(cf.gen_vectors(1, ct.default_dim),
                                                          ct.default_float_vec_field_name, ct.default_search_params,
                                                          ct.default_limit, check_task=CheckTasks.check_nothing)
                return search_res, is_succ

            def loop_search():
                """ continuously search """
                while True:
                    do_search()

            threading.Thread(target=loop_search, args=(), daemon=True).start()

            # scale in querynodes 
            mic.upgrade(release_name, {'spec.components.queryNode.replicas': scale_querynode}, constants.NAMESPACE)
            mic.wait_for_healthy(release_name, constants.NAMESPACE)
            wait_pods_ready(constants.NAMESPACE, f"app.kubernetes.io/instance={release_name}")
            log.debug("Scale out querynode success")

            time.sleep(60)
            log.debug("Scale out test finished")

            log.debug(f"total: {do_search.total}, succ: {do_search.succ}, fail: {do_search.fail}")

        except Exception as e:
            raise Exception(str(e))

        finally:
            log.info(f'Test finished')
```

### Milvus Log
**collection_name: scale_out_Kiyo6Gcq**
**Client error logs:**
```markdown
[2022-05-05 16:28:46,096 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.03470059099412844, 0.10317665810060968, 0.02934295892268309, 0.11961911489949086, 0.13840554633958194, 0.05071200981164083, 0.07707935888049891, 0.009374520551453646, 0.08102587955671664, 0.13731513513271976, 0.11834921045099361, 0.026205991954019382, 0.06915817069442196, 0.0016373746021157807,......, kwargs: {} (api_request.py:55)
[2022-05-05 16:28:51,242 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 672, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 21, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 481, in search
    return self._execute_search_requests(requests, timeout, **_kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 428, in _execute_search_requests
    raise pre_err
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 419, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=23, reason=Search 29 failed, reason cannot find segment 432993395697778690 in QueryNode err %!w(<nil>))>
 (api_request.py:35)
[2022-05-05 16:28:51,243 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=23, reason=Search 29 failed, reason cannot find segment 432993395697778690 in QueryNode err %!w(<nil>))> (api_request.py:36)
```

**Comment by ThreadDao** [comment]
If I do nothing while scaling, I can successfully search after scaling

**Comment by yah01** [comment]
The reason should be the same as #16644, working on it
#16770 and #16785 will fix this

**Comment by ThreadDao** [comment]
Fixed `master-20220506-67ce206c`
